packages/target-ethers-v5/src/codegen/index.ts

```diff
@@ -178,34 +181,44 @@ export function codegenContractFactory(contract: Contract, abi: any, bytecode?:
     connect(signer: Signer): ${contract.name}${FACTORY_POSTFIX} {
       return super.connect(signer) as ${contract.name}${FACTORY_POSTFIX};
     }
-    static connect(address: string, signerOrProvider: Signer | Provider): ${contract.name} {
-      return new Contract(address, _abi, signerOrProvider) as ${contract.name};
-    }
+    static readonly bytecode = "${bytecode.bytecode}";
+    ${body}
   }
 
-  const _abi = ${JSON.stringify(abi, null, 2)};
-
-  const _bytecode = "${bytecode.bytecode}";
-
   ${generateLibraryAddressesInterface(contract, bytecode)}
   `
 }
 
 export function codegenAbstractContractFactory(contract: Contract, abi: any): string {
+  const { body, header } = codegenCommonContractFactory(contract, abi)
   return `
   import { Contract, Signer } from "ethers";
   import { Provider } from "@ethersproject/providers";
-
-  import type { ${contract.name} } from "../${contract.name}";
+  import type { ${contract.name}, ${contract.name}Interface } from "../${contract.name}";
+  ${header}
 
   export class ${contract.name}${FACTORY_POSTFIX} {
-    static connect(address: string, signerOrProvider: Signer | Provider): ${contract.name} {
-      return new Contract(address, _abi, signerOrProvider) as ${contract.name};
-    }
+    ${body}
   }
+  `
+}
+
+function codegenCommonContractFactory(contract: Contract, abi: any): { header: string; body: string } {
+  const header = `
+  import { Interface } from '@ethersproject/abi';
 
   const _abi = ${JSON.stringify(abi, null, 2)};
-  `
+  `.trim()
+  const body = `
+    static readonly abi = _abi;
+    static get interface(): ${contract.name}Interface {
+      return new Interface(_abi) as ${contract.name}Interface;
+    }
```

Comment:
I think the `.createInterface()` function would be more ergonomic, especially with abstract factories.

Reply:
@krzkaczor WDYT? Especially in the light of "not adding helpers" but mainly typings (https://github.com/ethereum-ts/TypeChain/issues/370#issuecomment-816525127), the idea to use the same API for abstract and concrete contract factories seems nice, on the other hand it may be overkill / confuse users to make them similar.